src/hooks/useNetworkSwitch.ts

```diff
@@ -21,16 +21,25 @@ export const useNetworkSwitch = ({ onSelectNetworkCallback }: UseNetworkSwitchPr
   const selectNetwork = useCallback(
     (optionChainId?: ChainId) => {
       if (optionChainId === undefined || (optionChainId === chainId && !unsupportedChainIdError)) return
+      let changeChainId = new Promise(() => {})
       if (!account && !unsupportedChainIdError && connector instanceof CustomNetworkConnector)
-        connector.changeChainId(optionChainId)
+        changeChainId = connector.changeChainId(optionChainId)
       else if (!account && unsupportedChainIdError && connector instanceof CustomNetworkConnector)
-        connector.switchUnsupportedNetwork(NETWORK_DETAIL[optionChainId])
+        changeChainId = connector.switchUnsupportedNetwork(NETWORK_DETAIL[optionChainId])
       else if (connector instanceof InjectedConnector)
-        switchOrAddNetwork(NETWORK_DETAIL[optionChainId], account || undefined)
+        changeChainId = switchOrAddNetwork(NETWORK_DETAIL[optionChainId], account || undefined)
       else if (connector instanceof CustomWalletLinkConnector)
-        connector.changeChainId(NETWORK_DETAIL[optionChainId], account || undefined)
+        changeChainId = connector.changeChainId(NETWORK_DETAIL[optionChainId], account || undefined)
 
       if (onSelectNetworkCallback) onSelectNetworkCallback()
+      changeChainId.then(() => {
+        if (
+          !SWPR[optionChainId] &&
+          // check if the URL includes any of these
+          ['/pools', '/rewards'].reduce((state, str) => state || window.location.href.includes(str), false)
```

Comment:
Just thinking, maybe you can use `useLocation` and `useNavigate` hooks

Reply:
you should use `useNavigate` hook here

Reply:
the code doesn't like useNavigate here. This is not a react function.
Believe me, I would use that if I got that to work.
I'll add more details in a few minutes